api/cluster/controller.go

```diff
@@ -223,10 +223,19 @@ func (c *controller) Deploy(ctx context.Context, modelService *models.Service) (
 		return nil, errors.Wrapf(err, fmt.Sprintf("%v (%s)", ErrUnableToCreateInferenceService, isvcName))
 	}
 
-	s, err := c.kserveClient.InferenceServices(modelService.Namespace).Create(spec)
+	// check the cluster to see if the inference service has already been deployed
+	s, err := c.kserveClient.InferenceServices(modelService.Namespace).Get(modelService.Name, metav1.GetOptions{})
 	if err != nil {
-		log.Errorf("unable to create inference service %s: %v", isvcName, err)
-		return nil, errors.Wrapf(err, fmt.Sprintf("%v (%s)", ErrUnableToCreateInferenceService, isvcName))
+		if !kerrors.IsNotFound(err) {
```

Comment:
why is there an error if its found? anyway can it be rephrase so that there are no double negatives?

```
if kerrors.IsNotFound(err) {
   s, err = c.kserveClient.InferenceServices(modelService.Namespace).Create(spec)...
} else {
   ... errors.Wrapf(err, "unable to check status of inference service: %s")...
}
```

Reply:
> why is there an error if its found?

If it is found there won't be an error, the else block below, will be executed instead:
```go
} else {
	log.Infof("found existing inference service %s; skipping its creation", isvcName)
}
```

> anyway can it be rephrase so that there are no double negatives? 

Ah actually yeah it can 😅 I was initially avoiding it to use one less else-block but I agree it's probably better to avoid the double negative.